### PR TITLE
Improve comment for java-doc

### PR DESCRIPTION
**Pull request by alallema**

### CODE REVIEW

File: src/main/java/com/meilisearch/sdk/Documents.java [hunk 1]
```diff
@@ -6,7 +6,11 @@
 import com.meilisearch.sdk.model.Task;
 import java.util.List;
 
-/** Wrapper around HttpClient class to use for Meilisearch documents */
+/**
+ * Wrapper around HttpClient class to use for Meilisearch documents
+ *
+ * <p>https://docs.meilisearch.com/reference/api/documents.html
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/Documents.java:12]:
```suggestion
 * <p>https://docs.meilisearch.com/reference/api/documents.html</p>
```

File: src/main/java/com/meilisearch/sdk/IndexesHandler.java [hunk 2]
```diff
@@ -4,7 +4,11 @@
 import com.meilisearch.sdk.model.Task;
 import java.util.HashMap;
 
-/** Wrapper around the HttpClient class to ease usage for Meilisearch indexes */
+/**
+ * Wrapper around HttpClient class to use for Meilisearch indexes
+ *
+ * <p>https://docs.meilisearch.com/reference/api/indexes.html
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/IndexesHandler.java:10]:
```suggestion
 * <p>https://docs.meilisearch.com/reference/api/indexes.html</p>
```

File: src/main/java/com/meilisearch/sdk/InstanceHandler.java [hunk 3]
```diff
@@ -4,6 +4,11 @@
 import com.meilisearch.sdk.model.IndexStats;
 import com.meilisearch.sdk.model.Stats;
 
+/**
+ * Wrapper around HttpClient class to use for Meilisearch
+ *
+ * <p>https://docs.meilisearch.com/reference/api/keys.html
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/InstanceHandler.java:10]:
```suggestion
 * <p>https://docs.meilisearch.com/reference/api/keys.html</p>
```

File: src/main/java/com/meilisearch/sdk/KeysHandler.java [hunk 4]
```diff
@@ -7,7 +7,7 @@
 /**
  * Wrapper around HttpClient class to use for Meilisearch keys
  *
- * <p>Refer https://docs.meilisearch.com/reference/api/keys.html
+ * <p>https://docs.meilisearch.com/reference/api/keys.html
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/KeysHandler.java:10]:
```suggestion
 * <p>https://docs.meilisearch.com/reference/api/keys.html</p>
```

File: src/main/java/com/meilisearch/sdk/Search.java [hunk 5]
```diff
@@ -3,7 +3,11 @@
 import com.meilisearch.sdk.exceptions.MeilisearchException;
 import com.meilisearch.sdk.model.SearchResult;
 
-/** Search Object for searching on indexes */
+/**
+ * Wrapper around HttpClient class to use for Meilisearch search
+ *
+ * <p>https://docs.meilisearch.com/reference/api/search.html
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/Search.java:9]:
```suggestion
 * <p>https://docs.meilisearch.com/reference/api/search.html</p>
```

File: src/main/java/com/meilisearch/sdk/Search.java [hunk 6]
```diff
@@ -3,7 +3,11 @@
 import com.meilisearch.sdk.exceptions.MeilisearchException;
 import com.meilisearch.sdk.model.SearchResult;
 
-/** Search Object for searching on indexes */
+/**
+ * Wrapper around HttpClient class to use for Meilisearch search
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/Search.java:7]:
```suggestion
 * Search Object for searching on indexes
```

File: src/main/java/com/meilisearch/sdk/InstanceHandler.java [hunk 7]
```diff
@@ -4,6 +4,11 @@
 import com.meilisearch.sdk.model.IndexStats;
 import com.meilisearch.sdk.model.Stats;
 
+/**
+ * Wrapper around HttpClient class to use for Meilisearch
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/InstanceHandler.java:8]: Are you sure about this comment?

File: src/main/java/com/meilisearch/sdk/SettingsHandler.java [hunk 8]
```diff
@@ -9,7 +9,7 @@
 /**
  * Settings Handler for manipulation of an Index {@link Settings}
  *
- * <p>Refer https://docs.meilisearch.com/reference/api/settings.html
+ * <p>https://docs.meilisearch.com/reference/api/settings.html
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/SettingsHandler.java:12]:
```suggestion
 * <p>https://docs.meilisearch.com/reference/api/settings.html</p>
```

File: src/main/java/com/meilisearch/sdk/TasksHandler.java [hunk 9]
```diff
@@ -9,7 +9,7 @@
 /**
  * Wrapper around HttpClient class to use for MeiliSearch tasks
  *
- * <p>Refer https://docs.meilisearch.com/reference/api/tasks.html
+ * <p>https://docs.meilisearch.com/reference/api/tasks.html
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/TasksHandler.java:12]:
```suggestion
 * <p>https://docs.meilisearch.com/reference/api/tasks.html</p>
```

File: src/main/java/com/meilisearch/sdk/exceptions/APIError.java [hunk 10]
```diff
@@ -5,10 +5,10 @@
 import lombok.Setter;
 import lombok.experimental.Accessors;
 
+/** This class wraps errors sent by Meilisearch API */
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/exceptions/APIError.java:8]:
```suggestion
/** Meilisearch API error wrapper */
```

```suggestion
/** Wrapper around Meilisearch API errors */
```

File: src/main/java/com/meilisearch/sdk/exceptions/MeilisearchCommunicationException.java [hunk 11]
```diff
@@ -3,9 +3,9 @@
 import lombok.Getter;
 import lombok.Setter;
 
+/** This class wraps MeilisearchExceptions dealing with Communication errors */
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/exceptions/MeilisearchCommunicationException.java:6]:
```suggestion
/** Wrapper arround MeilisearchExceptions dealing with Communication errors */
```

File: src/main/java/com/meilisearch/sdk/exceptions/MeilisearchException.java [hunk 12]
```diff
@@ -3,9 +3,9 @@
 import lombok.Getter;
 import lombok.Setter;
 
+/** This is a generic class for Meilisearch Exception handling */
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/exceptions/MeilisearchException.java:6]:
```suggestion
/** Generic class for Meilisearch Exception handling */
```

File: src/main/java/com/meilisearch/sdk/exceptions/MeilisearchTimeoutException.java [hunk 13]
```diff
@@ -3,9 +3,9 @@
 import lombok.Getter;
 import lombok.Setter;
 
+/** This class wraps MeilisearchExceptions dealing with Timeout errors */
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/exceptions/MeilisearchTimeoutException.java:6]:
```suggestion
/** Wrapper around MeilisearchExceptions dealing with Timeout errors */
```

File: src/main/java/com/meilisearch/sdk/model/SearchResult.java [hunk 14]
```diff
@@ -7,7 +7,7 @@
 import lombok.ToString;
 
 /**
- * Data Structure Result of `search`
+ * Data Structure of Meilisearch response for Result of Search
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/model/SearchResult.java:10]:
```suggestion
 * Data Structure of Meilisearch response for results of search
```

**Review by bidoubiwa — CHANGES_REQUESTED** [review]

File: src/main/java/com/meilisearch/sdk/Documents.java (hunk 1 above)
Comment (alallema) [reply src/main/java/com/meilisearch/sdk/Documents.java:12]: (https://meilisearch.slack.com/archives/D01T5TK2XN2/p1668076946171329)
https://checkstyle.sourceforge.io/config_javadoc.html#JavadocParagraph

File: src/main/java/com/meilisearch/sdk/IndexesHandler.java [hunk 15]
```diff
@@ -4,7 +4,11 @@
 import com.meilisearch.sdk.model.Task;
 import java.util.HashMap;
 
-/** Wrapper around the HttpClient class to ease usage for Meilisearch indexes */
+/**
+ * Wrapper around HttpClient class to use for Meilisearch indexes
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/IndexesHandler.java:8]: Is it necessary to say that it wraps httpClient? Woulnd't it not be better to just state that it is the `Indexes API` ?

```suggestion
 * Meilisearch Indexes class
```

```suggestion
 * Indexes class covering all the methods related to an index
```

File: src/main/java/com/meilisearch/sdk/HttpClient.java [hunk 16]
```diff
@@ -13,7 +13,7 @@
 import com.meilisearch.sdk.json.JsonHandler;
 import java.util.Collections;
 
-/** The HTTP requests for the different functions to be done through Meilisearch */
+/** HTTP client to be used for the different functions to be performed through Meilisearch */
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/HttpClient.java:16]:
```suggestion
/** HTTP client used for API calls to Meilisearch */
```

I don't think it ads value to add more information. On the contrary, it confuses more.

File: src/main/java/com/meilisearch/sdk/Documents.java [hunk 17]
```diff
@@ -6,7 +6,11 @@
 import com.meilisearch.sdk.model.Task;
 import java.util.List;
 
-/** Wrapper around HttpClient class to use for Meilisearch documents */
+/**
+ * Documents class covering all the methods related to documents
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/Documents.java:10]:
```suggestion
 * Class covering the Meilisearch Document API
```

File: src/main/java/com/meilisearch/sdk/IndexesHandler.java [hunk 18]
```diff
@@ -4,7 +4,11 @@
 import com.meilisearch.sdk.model.Task;
 import java.util.HashMap;
 
-/** Wrapper around the HttpClient class to ease usage for Meilisearch indexes */
+/**
+ * IndexesHandler class covering all the methods related to an index
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/IndexesHandler.java:8]: Documents, which covers the documents API, is called `documents` but indexes, is cal.ed `IndexesHandler`. Is there a reason?

```suggestion
 * Class covering the Meilisearch Index API.
```

File: src/main/java/com/meilisearch/sdk/KeysHandler.java [hunk 19]
```diff
@@ -5,9 +5,9 @@
 import com.meilisearch.sdk.model.Result;
 
 /**
- * Wrapper around HttpClient class to use for Meilisearch keys
+ * Keys class covering all the methods related to keys
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/KeysHandler.java:8]:
```suggestion
 * Class covering the Meilisearch Key API
```

File: src/main/java/com/meilisearch/sdk/Search.java [hunk 20]
```diff
@@ -3,7 +3,11 @@
 import com.meilisearch.sdk.exceptions.MeilisearchException;
 import com.meilisearch.sdk.model.SearchResult;
 
-/** Search Object for searching on indexes */
+/**
+ * Search Object for searching on indexes
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/Search.java:7]:
```suggestion
 * Class used for searching on Meilisearch indexes
```

File: src/main/java/com/meilisearch/sdk/InstanceHandler.java [hunk 21]
```diff
@@ -4,6 +4,11 @@
 import com.meilisearch.sdk.model.IndexStats;
 import com.meilisearch.sdk.model.Stats;
 
+/**
+ * Wrapper around HttpClient class to get Meilisearch instance information
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/InstanceHandler.java:8]:
```suggestion
 * Class providing information on the Meilisearch instance
```

File: src/main/java/com/meilisearch/sdk/SettingsHandler.java [hunk 22]
```diff
@@ -9,7 +9,7 @@
 /**
  * Settings Handler for manipulation of an Index {@link Settings}
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/SettingsHandler.java:10]:
```suggestion
 * Class covering the Meilisearch Settings API {@link Settings}
```

File: src/main/java/com/meilisearch/sdk/TasksHandler.java [hunk 23]
```diff
@@ -7,9 +7,9 @@
 import java.util.Date;
 
 /**
- * Wrapper around HttpClient class to use for MeiliSearch tasks
+ * Task class covering all the methods related to tasks
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/TasksHandler.java:10]:
```suggestion
 * Class covering the Meilisearch Task API
```

File: src/main/java/com/meilisearch/sdk/model/IndexStats.java [hunk 24]
```diff
@@ -3,6 +3,11 @@
 import java.util.Map;
 import lombok.Getter;
 
+/**
+ * Data structure of Meilisearch response for the Stats of an Index
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/model/IndexStats.java:7]:
```suggestion
 * Stats data structure of a Meilisearch Index
```

File: src/main/java/com/meilisearch/sdk/model/SearchResult.java [hunk 25]
```diff
@@ -7,7 +7,7 @@
 import lombok.ToString;
 
 /**
- * Data Structure Result of `search`
+ * Data Structure of Meilisearch response for results of search
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/model/SearchResult.java:10]:
```suggestion
 * Meilisearch search response data structure
```

File: src/main/java/com/meilisearch/sdk/model/Settings.java [hunk 26]
```diff
@@ -6,7 +6,7 @@
 import lombok.experimental.Accessors;
 
 /**
- * Data Structure for the Settings
+ * Data Structure of Meilisearch response for the Settings
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/model/Settings.java:9]:
```suggestion
 * Meilisearch settings data structure
```

File: src/main/java/com/meilisearch/sdk/model/Stats.java [hunk 27]
```diff
@@ -5,7 +5,7 @@
 import lombok.Getter;
 
 /**
- * Data structure for the stats
+ * Data structure of Meilisearch response for the Stats
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/model/Stats.java:8]:
```suggestion
 * Meilisearch stats data structure
```

File: src/main/java/com/meilisearch/sdk/model/TaskDetails.java [hunk 28]
```diff
@@ -3,6 +3,7 @@
 import java.util.Map;
 import lombok.Getter;
 
+/** Data structure for the Details field in a Task */
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/model/TaskDetails.java:6]:
```suggestion
/** Task details data structure */
```

**Review by bidoubiwa — CHANGES_REQUESTED** [review]

File: src/main/java/com/meilisearch/sdk/model/TypoTolerance.java [hunk 29]
```diff
@@ -6,7 +6,7 @@
 import lombok.experimental.Accessors;
 
 /**
- * Data structure for the Typo tolerance setting
+ * Data structure of Meilisearch response for the Typo Tolerance Setting
```
Comment (bidoubiwa) [inline src/main/java/com/meilisearch/sdk/model/TypoTolerance.java:9]:
```suggestion
 * Typo Tolerance setting data structure
```

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥

**Comment by alallema** [comment]
bors merge

**Pull request merged**